deps/rabbit/src/rabbit_auth_backend_internal.erl

```diff
@@ -234,12 +234,9 @@ add_user_sans_validation(Username, Password, ActingUser) ->
         throw:{error, {user_already_exists, _}} = Error ->
             rabbit_log:warning("Failed to add user '~s': the user already exists", [Username]),
             throw(Error);
-        throw:Error ->
+        Class:Error ->
             rabbit_log:warning("Failed to add user '~s': ~p", [Username, Error]),
-            throw(Error);
-        exit:Error ->
-            rabbit_log:warning("Failed to add user '~s': ~p", [Username, Error]),
-            exit(Error)
+            erlang:Class(Error)
```

Comment:
We might want to consider using `erlang:raise/3` to have the correct stacktrace for the original exception.

Reply:
True. I'll submit an update (later) to include stacktrace in final exception.

It is something which I find still being used quite sparingly since its introduction in otp-21. Do we make it a convention to always, or more often than not, use/include it in future? (questions from a contributor 😅 )

Reply:
@Ayanda-D I've not received any objections so please go ahead with such a patch.